### DISCUSSION TITLE
How to get device instance path?

### DISCUSSION THREAD

**Discussion by gongjianbo** [libusb/libusb#1335] [Q&A]
I need some platform related descriptions. On Windows, the format is like "\VID_C0F4&PID_01F5&MI_01&Col01\9&c491a5&0&0000".

**Comment by diabolo38** [comment]
why would you need this ? Unless doing very special or wiring your own inf file or alike it's not something to care about.

BTW find the device in the device manager , find it , check details, porperty 
![image](https://github.com/libusb/libusb/assets/8239335/182f8637-662c-445d-b973-1130924e16c5)
 
they may be multiple possible option.

if your devices as no device driver attached then should be easy to spot
Device manager show device by connection may help but coudl be painfull as you have to follow/Expand  the tree all ways down to the device

**Reply by gongjianbo** [reply]
Because the existing code uses this path, if there is no available interface, then I can only modify the source code to export the path obtained in libusb.The old code was based on winusb, and the new code needed to be cross-platform, so it was switched to libusb.

**Comment by diabolo38** [comment]
hum note i think you can claim and use the overall device (if is as 1 n  or no interface)   just using the USB/VID_XXX/PID_XXXX witthout any /MIXX_etc that refer to interface number version etc ....
you likely better to review and change this partsof the original code as it will hardly map to similar thing in Linux and mac 

good luck ;)

**Reply by gongjianbo** [reply]
I seem to have found a way, through the interface in libusbi.h:
     
winusb_device_priv *priv = (winusb_device_priv *)(usbi_get_device_priv(device));
fprintf(stderr, "Device Priv: %p\n", priv);
 if (priv) {
      fprintf(stderr, "Device Path: %s\n", priv->path);
}

will print : \\?\USB#VID_048D&PID_C103#7&1B8A7FB1&0&2#{A5DCBF10-6530-11D2-901F-00C04FB951ED}

**Comment by vovkos** [comment]
`winusb_device_priv` and `usbi_get_device_priv` are not public `libusb` APIs and could change without notice. Relying on such internal structures or methods is generally considered a bad practice (i.e., your code can suddenly stop working with a newer version of `libusb`).

This PR https://github.com/libusb/libusb/pull/1314 takes care of what you need. You can upvote this PR to increase its chances of getting merged in.

**Comment by mcuee** [comment]
Kind of related discussion -- not the same but kind of related, just for reference.
https://community.osr.com/discussion/294108/get-path-for-device-being-driven-from-inside-kernel-mode-driver
